presto-verifier/src/main/java/com/facebook/presto/verifier/framework/AbstractVerification.java

```diff
@@ -351,6 +357,12 @@ protected static String formatSql(Statement statement)
         if (determinismAnalysis.isPresent() && determinismAnalysis.get().isNonDeterministic()) {
             return Optional.of(NON_DETERMINISTIC);
         }
+        if (queryException.isPresent() &&
+                queryException.get().getQueryStage().equals(CHECKSUM) &&
+                queryException.get().getPrestoErrorCode().isPresent() &&
+                queryException.get().getPrestoErrorCode().get().equals(COMPILER_ERROR)) {
```

Comment:
I'm wondering if this logic should apply to any kind of failure that occurred in checksum query for the control run. The other day we saw checksum query failing with "value too large".

Reply:
I think we should more finely handle the different cases. In case there are better solution for a different type of error, we shouldn't blindly skipping them.

Reply:
@caithagoras A workaround for "value too large" error could be computing checksums on primitive columns separately and complex columns one at a time and comparing checksums that can be computed and ignoring the ones which cannot. What do you think?